### PR TITLE
feat(hotkey): add an interface to query hotkey on the specified partition

### PR DESCRIPTION
**Pull request by Smityz**

### CODE REVIEW

**Review by acelyc111 — APPROVED** [review]

**Review by neverchanje — DISMISSED** [review]

File: src/server/hotkey_collector.cpp [hunk 1]
```diff
@@ -168,6 +168,7 @@ inline void hotkey_collector::change_state_by_result()
     case hotkey_collector_state::FINE_DETECTING:
         if (!_result.hot_hash_key.empty()) {
             change_state_to_finished();
+            derror_replica("Find the hotkey: {}", _result.hot_hash_key);
```
Comment (levy5307) [inline src/server/hotkey_collector.cpp:171]: why did you use derror_replica? what about dwarn_replica?

Comment (Smityz) [reply src/server/hotkey_collector.cpp:171]: In pegasus, `error` is higher than `dwarn`

File: src/server/hotkey_collector.cpp [hunk 2]
```diff
@@ -272,6 +276,20 @@ void hotkey_collector::on_stop_detect(dsn::replication::detect_hotkey_response &
     ddebug_replica(hint);
 }
 
+void hotkey_collector::query_result(dsn::replication::detect_hotkey_response &resp)
+{
+    if (_state != hotkey_collector_state::FINISHED) {
+        resp.err = dsn::ERR_BUSY;
+        std::string hint = fmt::format("hotkey is detecting now, now state: {}",
+                                       dsn::enum_to_string(_hotkey_type));
+        resp.err_hint = hint;
```
Comment (hycdong) [inline src/server/hotkey_collector.cpp:285]: I recommend use `__set_err_hint` instead.

File: src/server/hotkey_collector.cpp [hunk 3]
```diff
@@ -272,6 +276,20 @@ void hotkey_collector::on_stop_detect(dsn::replication::detect_hotkey_response &
...
+        ddebug_replica(hint);
+    } else {
+        resp.err = dsn::ERR_OK;
+        resp.hotkey_result = _result.hot_hash_key;
```
Comment (hycdong) [inline src/server/hotkey_collector.cpp:289]: I recommend use `__set_hotkey_result` instead.

**Review by acelyc111 — APPROVED** [review]

**Review by hycdong — APPROVED** [review]

**Pull request merged**